app/code/community/Bolt/Boltpay/Helper/Data.php

```diff
@@ -98,13 +96,39 @@ public function getBoltCallbacks($checkoutType, $isVirtualQuote = false)
     }
 
     /**
-     * @param      $checkoutType
+     * @param string $checkoutType
      * @param bool $isVirtualQuote
      *
      * @return string
+     *
+     * @throws Mage_Core_Model_Store_Exception
      */
     public function buildOnCheckCallback($checkoutType, $isVirtualQuote = false)
     {
+        if ( $checkoutType === Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_ADMIN ) {
+            $checkoutTokenUrl = $this->getMagentoUrl("adminhtml/sales_order_create/create/checkoutType/$checkoutType", array(), true);
+        } else {
+            $checkoutTokenUrl = $this->getMagentoUrl("boltpay/order/create/checkoutType/$checkoutType");
+        }
+
+        $ajaxCall = "
+            new Ajax.Request('$checkoutTokenUrl', {
+                method:'post',
+                parameters: '',
+                onSuccess: function(response) {
+                    if(response.responseJSON.error) {                                                        
+                        rejectPromise(response.responseJSON.error_messages);
+                        
+                        // BoltCheckout is currently not doing anything reasonable to alert the user of a problem, so we will do something as a backup
+                        alert(response.responseJSON.error_messages);
```

Comment:
What does this alert look like, do you have a screenshot?
I think the lack of external messaging is an intentional decision, cc:ing @daisy1754 here for his opinion.

Reply:
yeah let's not do alert - alert is such an unpolished UI. what are the example of errors that "please try again" don't fit well? we can chat with product to define some error type. As Matt said not showing error message as is probably intentional decision, as often these errors are not necessarily user friendly

Reply:
I considered a few things and had questions for one of them.

@daisy1754, @matt-thomason  
- What does Bolt now do by default?  When this code was first written, Bolt did nothing so, I thought to keep it.  But, not showing anything is an option.

- Regarding unpolished, it is actually the same interface used by standard magento, so it will appear to be common to the end user.

- We have our own pop-up that we can use.  As is, this is a let's make Mike A. happy PR on a tight timeline.  We can use the same popup that we use for refunds if there is suitable message.

Reply:
ok.  Then we will give no indication to the merchant at this time.  We can revisit this at a future date if we decide we wish to inform the merchant with a user-friendly popup.

Reply:
oh i thought this PR is for firecheckout. for admin I think it's ok to use alert but firecheckout I'd vote against. we actually had merchant complained us in the past about using alert saying because it's not sophisticated ux